WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m

```diff
@@ -606,48 +618,123 @@ - (UITableViewCell *)configureFeaturedImageCellForIndexPath:(NSIndexPath *)index
+        }];
+    }
+    _geoLocationSwitchCell.on = self.post.geolocation != nil || [self.locationService locationServiceRunning];
+    return _geoLocationSwitchCell;
+}
+
+- (PostGeolocationCell *)postGeoLocationCell {
+    if (!_postGeoLocationCell) {
+        _postGeoLocationCell = [[PostGeolocationCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:nil];
+        _postGeoLocationCell.tag = PostSettingsRowGeolocationMap;
+    }
+    Coordinate *coordinate = self.post.geolocation;
+    NSString *address = NSLocalizedString(@"Finding your location...", @"Geo-tagging posts, status message when geolocation is found.");
+    if (coordinate) {
         CLLocation *postLocation = [[CLLocation alloc] initWithLatitude:coordinate.latitude longitude:coordinate.longitude];
-        NSString *address;
-        if ([[LocationService sharedService] hasAddressForLocation:postLocation]) {
-            address = [LocationService sharedService].lastGeocodedAddress;
+        if ([self.locationService hasAddressForLocation:postLocation]) {
+            address = self.locationService.lastGeocodedAddress;
         } else {
             address = NSLocalizedString(@"Looking up address...", @"Used with posts that are geo-tagged. Let's the user know the the app is looking up the address for the coordinates tagging the post.");
-            [[LocationService sharedService] getAddressForLocation:postLocation
+            __weak __typeof__(self) weakSelf = self;
+            [self.locationService getAddressForLocation:postLocation
                                                         completion:^(CLLocation *location, NSString *address, NSError *error) {
-                                                            [self.tableView reloadData];
+                                                            [weakSelf.tableView reloadData];
                                                         }];
-
+            
         }
-        [geoCell setCoordinate:self.post.geolocation andAddress:address];
-        cell = geoCell;
-        cell.tag = PostSettingsRowGeolocationMap;
+    }
+    [_postGeoLocationCell setCoordinate:coordinate andAddress:address];
+    return _postGeoLocationCell;
+}
+
+- (WPTableViewCell *)geoLocationActivityCell {
+    if (!_geoLocationActivityCell) {
+        _geoLocationActivityCell = [[WPTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:nil];
+        _geoLocationActivityCell.accessoryType = UITableViewCellAccessoryNone;
+        [WPStyleGuide configureTableViewCell:_geoLocationActivityCell];
+        UIActivityIndicatorView *spinner = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
+        _geoLocationActivityCell.accessoryView = spinner;
+        [spinner startAnimating];
+        _geoLocationActivityCell.textLabel.text = NSLocalizedString(@"Finding your location...", @"Geo-tagging posts, status message when geolocation is found.");
```

Comment:
Would it make sense to call a method to return the localized string rather than duplicate it here?